lib/extend.js

```diff
@@ -111,18 +112,21 @@ function extend(opts) {
     if (!_.isArray(options.source))
       throw new Error('source must be an array');
 
-    var props = combineJSON(options.source, true, function Collision(opts) {
-      if (options.log) {
-        var str = `Collision detected at: ${opts.path.join('.')}! Original value: ${opts.target[opts.key]}, New value: ${opts.copy[opts.key]}`;
-        if (options.log === 'warn') {
-          console.warn(chalk.keyword('orange')(str));
-        } else if (options.log === 'error') {
-          throw new Error(str);
-        }
+    var ERROR_TYPE = 'Property Value Collisions';
```

Comment:
Maybe instead of these ERROR_TYPE strings we could actually subclass Error to have actually different Error types.  
https://gist.github.com/justmoon/15511f92e5216fa2624b

Reply:
Doing what Danny suggested for subclassing.

Also, I'm leaving the declaration in mid-code here because it doesn't apply to the whole file - just the calls immediately below it. Instead, I'm moving the declaration of the error group in exportPlatform to be only next to the areas it is used. It is actually the fact that it is defined the same in exportPlatform and in resolveObject that it works as expected. Let me know if you think that I should make this more clear somehow / pass this information in?